Small Cable Fix

Ryan180602 opened the pull request:
:cl:
bugfix: Making restraints shows progress bar.
bugfix: Doesn't deplete coil irregardless of completion of restraint making action.
/:cl:

smol fix

MuckerMayhem commented on code/modules/power/cable_coil.dm:
```diff
@@ -251,14 +251,14 @@
 	if (!isturf(user.loc))
 		to_chat(user, SPAN_WARNING("You don't have enough free space to do that."))
 		return
-	if (!use(15))
```
Won't you need to include a `use(15)` somewhere below so that the cable _is_ used when there is enough? If I'm reading this right, it just won't use any cable at all even if they have enough.

SierraKomodo replied:
`use(N)` will use the cable if it succeeds. `can_use()` is the one that won't actually use the cable.

SierraKomodo replied:
That said, it's better to put the actual `use()` call _after_ the do_after() in case the timer gets cancelled so that the cable doesn't disappear and make the pre-timer check `can_use()` instead.

SierraKomodo commented on code/modules/power/cable_coil.dm:
```diff
@@ -251,14 +251,14 @@
 		SPAN_ITALIC("You begin to wind the cable into crude restraints.")
 	)
-	if (!do_after(user, 10 SECONDS, src, DO_DEFAULT | DO_PUBLIC_PROGRESS))
+	if (!do_after(user, 10 SECONDS, usr, DO_DEFAULT | DO_PUBLIC_PROGRESS))
```
This should probably just be null instead. - `usr` is redundant because `user` is already being passed into this.

MuckerMayhem commented on code/modules/power/cable_coil.dm:
```diff
@@ -251,14 +251,14 @@
 		to_chat(user, SPAN_WARNING("You don't have enough free space to do that."))
 		return
-	if (!use(15))
+	if (src.get_amount() < 15)
```
```suggestion
	if (get_amount() < 15)
```
didn't catch this first time, `src.` is redundant.

Ryan180602 commented:
Why did I start a review.

Also, issues fixed.

Spookerton commented on code/modules/power/cable_coil.dm:
```diff
@@ -251,19 +251,20 @@
 		SPAN_ITALIC("\The [user] makes some crude cable restraints."),
 		SPAN_NOTICE("You finish making the restraints.")
 	)
+	src.use(15)
```
```suggestion
	use(15)
```

Spookerton commented on code/modules/power/cable_coil.dm:
```diff
@@ -251,19 +251,20 @@
 		to_chat(user, SPAN_WARNING("You don't have enough free space to do that."))
 		return
-	if (!use(15))
+	if (!src.can_use(15))
```
```suggestion
	if (!can_use(15))
```

Ryan180602 committed:
make cable restraint progress bar be visible

also makes them not use 15 cables even if action is interrupted
a


ryan180602 #32670 cablefix
